astropy/io/fits/hdu/compressed/tests/test_checksum.py

```diff
@@ -19,9 +19,9 @@ def test_compressed_image_data(self):
                 assert "DATASUM" in h2[0].header
                 assert h2[0].header["DATASUM"] == "0"
                 assert "CHECKSUM" in h2[1].header
-                assert h2[1].header["CHECKSUM"] == "ZeAbdb8aZbAabb7a"
+                assert h2[1].header["CHECKSUM"] == "D2GXD29VD2EVD29V"
                 assert "DATASUM" in h2[1].header
-                assert h2[1].header["DATASUM"] == "113055149"
+                assert h2[1].header["DATASUM"] == "2189405276"
```

Comment:
Does it make sense that both checksum and datasum have changed?

Reply:
I think checksum includes the data and the header so since the datasum has changed I think it makes sense that checksum changes too

Reply:
I see you mention this in the changelog entry:
> Fix the image checksum/datasum in ``CompImageHDU.header`` to be those for the
  image HDU instead of for the underlying binary table.

And the discussion I was remembering is #14753.
However I'm not convinced that CHECKSUM/DATASUM should be the ones of the (uncompressed) image.

https://github.com/astropy/astropy/blob/a32e8565405c88c333192ae4d3d3b43216fdd4be/docs/io/fits/appendix/history.rst#321-2014-03-04 mentions this in PyFITS 3.2.1:
> Fixed checksums on compressed images, so that the ZHECKSUM and ZDATASUM contain a checksum of the original image HDU, while CHECKSUM and DATASUM contain checksums of the compressed image HDU. This feature was supposed to be supported in 3.2, but the support was buggy.

This seems more logical to me.

Reply:
Regarding the changelog entry, I think we do want CompImageHDU.header (which is the decompressed/image header) to contain the image checksum. This should then get stored as ZHECKSUM on disk. And if one opens the file disabling the image compression, BinTableHDU.header will contain the BinTableHDU checksum, stored as CHECKSUM on disk. Does this make sense?

I'll double check that this is the actual behaviour though.